proxy/http/HttpConfig.h

```diff
@@ -736,6 +733,8 @@ struct OverridableHttpConfigParams {
   MgmtInt default_buffer_water_mark;
   MgmtInt slow_log_threshold;
 
+  OutboundConnTrack::TxnConfig outbound_conntrack;
```

Comment:
This is something I've discussed with Leif and others, to simplify the HTTP configuration structures. Rather than declaring the configuration for a subsystem here, it's declared in the subsystem and included via member variable here. Updates are then more localized. More importantly, when it is necessary to pass this configuration to the subsystem for a specific transaction, just this structure can be passed and therefore the subsystem doesn't have to include all of the HTTP configuration headers. This is simpler and removes one major cause of circular dependencies. Now the HTTP configuration depends on the subsystem, but not the reverse.

Reply:
👍